src/tasks/deployment-of-bridged-token-deployer.ts

```diff
@@ -0,0 +1,126 @@
+import { promises as fs } from "fs";
+
+import { task } from "hardhat/config";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+
+import {
+  computeProofs,
+  parseCsvFile,
+  DeploymentHelperDeployParams,
+  generateProposal,
+  constructorInput,
+  ContractName,
+} from "../ts";
+import { removeSplitClaimFiles, splitClaimsAndSaveToFolder } from "../ts/split";
+
+import { Args, Settings } from "./common-interfaces";
+import { defaultTokens } from "./ts/constants";
+import { defaultSafeDeploymentAddresses } from "./ts/safe";
+
+export const OUTPUT_FOLDER_GC = "./output/deployment-gc";
+
+const setupBridgedTokenDeployerTask: () => void = () => {
+  task(
+    "deployment-bridged-token-deployer",
+    "Generate the list of claims from a csv and reads settings from json and deploys the bridged token deployer on gnosis chain",
+  )
+    .addParam(
+      "claims",
+      "Path to the CSV file that contains the list of claims to generate.",
+    )
+    .addParam(
+      "settings",
+      "Path to the JSON file that contains the deployment settings.",
+    )
+    .setAction(generateDeployment);
+};
+
+async function generateDeployment(
+  { claims: claimCsv, settings: settingsJson }: Args,
+  hre: HardhatRuntimeEnvironment,
+): Promise<void> {
+  const { ethers } = hre;
+  const [deployer] = await ethers.getSigners();
+
+  const chainId = (await hre.ethers.provider.getNetwork()).chainId.toString();
+  if (chainId !== "100") {
+    throw new Error(
+      `This script must be run on gnosis chain. Found chainId ${chainId}`,
+    );
+  }
+
+  const inputSettings: Settings = JSON.parse(
+    await fs.readFile(settingsJson, "utf8"),
+  );
+  console.log(`Using deployer ${deployer.address}`);
+
+  console.log("Reading user claims for gnosis chain from file...");
+  const claims = await parseCsvFile(claimCsv);
+
+  console.log("Generating Merkle proofs...");
+  const { merkleRoot, claims: claimsWithProof } = computeProofs(claims);
+
+  const settings = {
+    ...inputSettings,
+    virtualCowToken: {
+      gnoPrice: inputSettings.gnoPrice,
+      nativeTokenPrice: inputSettings.nativeTokenPriceOnGnosisChain,
+      merkleRoot,
+      usdcToken: defaultTokens.usdc[chainId],
+      gnoToken: defaultTokens.gno[chainId],
+      wrappedNativeToken: defaultTokens.weth[chainId],
+    },
+  };
+
+  const { addresses } = await generateProposal(
+    settings,
+    defaultSafeDeploymentAddresses(chainId),
+    hre.ethers,
+  );
```

Comment:
A bit risky to use this function to get the addresses as there are many assumptions that are being made that aren't obvious.
For example, we are setting the xdai `merkleRoot`, but this would be wrong on the mainnet proposal, so we are betting that the Merkle root is unused (which it is—now). On the other way this is a very pragmatic way to do it, so let's go for it! However:
1. I'd add a comment about this before this call that explains at length that we are abusing this function.
2. I'd use `constants.HashZero` instead of the Merkle root to avoid misunderstandings.
Also, I expect `generateProposal to be extended with extra xdai parameters, so they should all be filled with fake data.